packages/task/src/browser/task-service.ts

```diff
@@ -506,8 +506,20 @@ export class TaskService implements TaskConfigurationClient {
             }
         }
 
-        const tasks = await this.getWorkspaceTasks(task._scope);
         const resolvedMatchers = await this.resolveProblemMatchers(task, customizationObject);
+        const runTaskOption: RunTaskOption = {
+            customization: { ...customizationObject, ...{ problemMatcher: resolvedMatchers } }
+        };
+
+        if (task.dependsOn) {
+            return this.runCompoundTask(task, runTaskOption);
+        } else {
+            return this.runTask(task, runTaskOption);
```

Comment:
Does it change error handling semantic here? It seems before the method will return `undefined` but now it throws? or `runTask` never throws?

Reply:
it catches errors at running a configuration [here](https://github.com/eclipse-theia/theia/blob/e31a23d6566ef4d2f92fb53f5280783b812e7de0/packages/task/src/browser/task-service.ts#L929-L932), but, taking into account that we have chaining of call at running task, I added the corresponding block: https://github.com/eclipse-theia/theia/pull/7696/commits/a5367f9314321de98ccf3894b4aaea4d2652d91e  
thank you!